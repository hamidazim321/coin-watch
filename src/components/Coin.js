import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import Styles from '../styles/Coin.module.css';

const Coin = ({ props }) => {
  const {
    symbol, rank, id,
  } = props;
  return (
    <div className={Styles.coin}>
      <NavLink to={`/Details/${id}`}>==</NavLink>
      <p className={Styles.name}>{symbol}</p>
      <p className={Styles.rank}>{`Ranking: ${rank}`}</p>
    </div>
  );
};

Coin.propTypes = {
  props: PropTypes.shape({
    symbol: PropTypes.string,
  }).isRequired,
  symbol: PropTypes.string.isRequired,
  rank: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
};

export default Coin;
