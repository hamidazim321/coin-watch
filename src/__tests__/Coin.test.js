import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import Coin from '../components/Coin';
import { mockCoins } from './__mocks__/mockCoins';

describe('Coin Card', () => {
  test('Should render correctly', () => {
    const coin = render(
      <MemoryRouter>
        <Coin props={mockCoins[0]} />
      </MemoryRouter>,
    );
    expect(coin).toMatchSnapshot();
  });
});
