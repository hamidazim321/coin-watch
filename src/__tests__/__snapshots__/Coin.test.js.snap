// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coin Card Should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="coin"
      >
        <a
          class="arrow"
          href="/Details/90"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M438.6 278.6c12.5-12.5 12.5-32.8 0-45.3l-160-160c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L338.8 224 32 224c-17.7 0-32 14.3-32 32s14.3 32 32 32l306.7 0L233.4 393.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0l160-160z"
            />
          </svg>
        </a>
        <p
          class="name"
        >
          BTC
        </p>
        <p
          class="rank"
        >
          Ranking: 1
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="coin"
    >
      <a
        class="arrow"
        href="/Details/90"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M438.6 278.6c12.5-12.5 12.5-32.8 0-45.3l-160-160c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L338.8 224 32 224c-17.7 0-32 14.3-32 32s14.3 32 32 32l306.7 0L233.4 393.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0l160-160z"
          />
        </svg>
      </a>
      <p
        class="name"
      >
        BTC
      </p>
      <p
        class="rank"
      >
        Ranking: 1
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
